presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -486,32 +458,20 @@ private void planDistribution(PlanRoot plan)
         // record output field
         stateMachine.setColumns(((OutputNode) outputStagePlan.getFragment().getRoot()).getColumnNames(), outputStagePlan.getFragment().getTypes());
 
-        PartitioningHandle partitioningHandle = outputStagePlan.getFragment().getPartitioningScheme().getPartitioning().getHandle();
-        OutputBuffers rootOutputBuffers = createInitialEmptyOutputBuffers(partitioningHandle)
-                .withBuffer(OUTPUT_BUFFER_ID, BROADCAST_PARTITION_ID)
-                .withNoMoreBufferIds();
-
         SplitSourceFactory splitSourceFactory = new SplitSourceFactory(splitSourceProvider);
         // build the stage execution objects (this doesn't schedule execution)
         SqlQueryScheduler scheduler = createSqlQueryScheduler(
-                stateMachine,
                 locationFactory,
-                outputStagePlan,
-                nodePartitioningManager,
-                nodeScheduler,
+                executionPolicy,
+                queryExecutor,
+                schedulerStats,
+                sectionStageExecutionFactory,
                 remoteTaskFactory,
                 splitSourceFactory,
                 stateMachine.getSession(),
```

Comment:
Totally unrelated to this PR. Since `createSqlQueryScheduler()` is a factory method, and we already passed `stateMachine` in, this seems kinda redundant.